Formula/k/k6.rb

```diff
@@ -14,7 +14,7 @@ class K6 < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "6c59c33536818d69e337bd36ce5b3e58054a019260d35027efae3b50ca7b56a5"
   end
 
-  depends_on "go" => :build
+  depends_on "go@1.22" => :build
```

Comment:
can you leave some issue ref as code comment in here?

Reply:
https://github.com/grafana/k6/issues/2474 basically was due to change in go 1.19. Brew build it with go 1.19 which was out, while k6 was still using 1.18 to build and release k6. We usually move to the next version after the first `.1` release and it usually happens that a new major go version gets released the week before a planned k6 release. And there really has never been any reason for us to jump to the next release. 

So while brew using latest versions of packages is likely a good default, we really would prefer to have to bump this for k6 every 6 months or so as we bump it internally.

In this case here I actualyl want this to be pinned to `1.23` now, but that gave lint errors previously as you can see. So I was testing if `@1.22` will give the same lint errors - which it seems not to.

Can we set it to 1.23 and not have the lint errors ? 
```suggestion
  depends_on "go@1.23" => :build # See https://github.com/grafana/k6/issues/2699 and https://github.com/grafana/k6/issues/2474
```

Reply:
You do realise, that Go 1.22 will not be supported anymore after 1.24 is released in few weeks (expected in February, currently in rc1, rc2 is expected tomorrow), see 
* https://go.dev/wiki/Go-Release-Cycle and
* https://github.com/Homebrew/homebrew-core/pull/201070

Go 1.23 was released in August (=6 months ago), with several minor releases, giving it plenty of time to polish any performance issues.

Please consider pinning it to Go 1.23 (which is also currently being used for building k6 in brew), until you find time to make k6 (and other grafana projects) compatible with Go 1.24 and later releases.